lang/cs/Org.Apache.REEF.Wake/Time/Runtime/RuntimeClock.cs

```diff
@@ -151,29 +152,43 @@ public override void Dispose()
         public void Run()
         {
             SubscribeHandlers();
-            _handlers.OnNext(new RuntimeStart(_timer.CurrentTime));
-            _handlers.OnNext(new StartTime(_timer.CurrentTime));
 
-            while (true)
+            var runtimeException = Optional<Exception>.Empty();
+            try
             {
-                lock (_schedule)
+                _handlers.OnNext(new RuntimeStart(_timer.CurrentTime));
+                _handlers.OnNext(new StartTime(_timer.CurrentTime));
+
+                while (true)
                 {
-                    if (IsIdle())
+                    lock (_schedule)
                     {
-                        _handlers.OnNext(new IdleClock(_timer.CurrentTime));
-                    }
-                    
-                    // Blocks and releases lock until it receives the next event
-                    Time alarm = GetNextEvent();
-                    ProcessEvent(alarm);
+                        if (IsIdle())
+                        {
+                            _handlers.OnNext(new IdleClock(_timer.CurrentTime));
+                        }
 
-                    if (alarm is StopTime)
-                    {
-                        break;
+                        // Blocks and releases lock until it receives the next event
+                        Time alarm = GetNextEvent();
+                        ProcessEvent(alarm);
+
+                        if (alarm is StopTime)
+                        {
+                            break;
+                        }
                     }
                 }
             }
-            _handlers.OnNext(new RuntimeStop(_timer.CurrentTime));
+            catch (Exception e)
+            {
+                runtimeException = Optional<Exception>.Of(e);
```

Comment:
Maybe wrap the exception received with an error message indicating where we caught it?